pandas/core/ops.py

```diff
@@ -1244,6 +1244,32 @@ def wrapper(left, right):
                                     index=left.index, name=res_name,
                                     dtype=result.dtype)
 
+        elif is_timedelta64_dtype(right) and not is_scalar(right):
+            # i.e. exclude np.timedelta64 object
+            # GH#22390 Unfortunately we need to special-case right-hand
+            # timedelta64 dtypes because numpy casts integer dtypes to
+            # timedelta64 when operating with timedelta64
+            if isinstance(right, np.ndarray):
+                # upcast to TimedeltaIndex before dispatching
```

Comment:
uhh this seems like a lot of special casing, any way to make this simpler

Reply:
The only simplification I considered but didn't use was something like:

```
def maybe_upcast_for_op(obj):
     if type(obj) is timedelta:
        return pd.Timedelta(obj)
    if isinstance(obj, np.ndarray) and is_timedelta64_dtype(obj):
        return pd.TimedeltaIndex(obj)
    return obj
```

which would go right after `get_op_result_name`.

Reply:
maybe re-route the scalars to a separate path. This is getting way special casey here.

Reply:
I'll implement the maybe_upcast idea and see if that is prettier